EuclideanGeometry/Foundation/Axiom/Linear/Line.lean

```diff
@@ -64,7 +64,30 @@ theorem same_extn_line_of_PM (A : P) (x y : Dir) (h : PM x y) : same_extn_line (
   · exact Or.inl Ray.source_lies_on
 
 theorem same_extn_line.eq_carrier_union_rev_carrier (ray ray' : Ray P) (h : same_extn_line ray ray') : ray.carrier ∪ ray.reverse.carrier = ray'.carrier ∪ ray'.reverse.carrier := by
-  sorry
+  ext p
+  simp only [Set.mem_union, Ray.in_carrier_iff_lies_on, pt_lies_on_line_from_ray_iff_vec_parallel]
+  let ⟨a, ha⟩ := pt_lies_on_line_from_ray_iff_vec_parallel.mp h.symm.2
+  let ⟨b, hb⟩ := vec_parallel_of_same_extn_line h.symm
+  constructor
+  · intro hc
+    rcases hc with ⟨c, hc⟩
+    use a + c * b
+    calc
+      VEC ray'.source p = VEC ray'.source ray.source + VEC ray.source p := (vec_add_vec _ _ _).symm
+      _ = a • ray'.toDir.toVec + c • ray.toDir.toVec := by rw [ha, hc]
+      _ = a • ray'.toDir.toVec + (c * b) • ray'.toDir.toVec := by
+        simp only [hb, Complex.real_smul, Complex.ofReal_mul, add_right_inj]
+        ring_nf
+      _ = (a + c * b) • ray'.toDir.toVec := (add_smul _ _ _).symm
+  · sorry
+  /-
+  Proof for the second case is almost the same with the first case.
+  I suppose it better to write a tactic for this kind of proof ?
+  (i.e., reducing arguments about `•` and `VEC`) 
+
+  Another issue is that, `Lean Infoview` doesn't show enough info about the type. 
+  e.g., there could be `Dir.toVec = b • Dir.toVec` in `Infoview`, but moving cursor to `Dir.toVec` doesn't tells me which is the argument of this function.
+  -/
```

Comment:
Thanks for your comments! 
1. For the tactic problem, I think we should fill this sorry first since we still need some time to write this tactic.
2. The info view is a serious problem. It does not unfold enough. maybe we can also write a tactic about this. This tactic seems to be easier to write than the first one !

Reply:
(by 2 I mean a tactic that forces the info view to show more information)